src/PrestaShopBundle/DataCollector/ConfigDataCollector.php

```diff
@@ -0,0 +1,316 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\DataCollector;
+
+use PrestaShop\PrestaShop\Core\Foundation\Version;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\DataCollector\LateDataCollectorInterface;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Component\VarDumper\Caster\ClassStub;
+
+/**
+ * Class ConfigDataCollector.
+ * This class is used to collect some data for the WebProfiler.
+ */
+class ConfigDataCollector extends DataCollector implements LateDataCollectorInterface
```

Comment:
Seems lie you replace the previous `Symfony\Component\HttpKernel\DataCollector\ConfigDataCollector` from the framework by this class, which makes sense

But it also seems like you copied most of its code which is not necessary and makes it harder to maintain, I think you should simply extend the class and override the `collect` method to add only the additional data

Reply:
Yes, that was my first attempt on this. But in the phpdoc of the Symfony ConfigDataCollector class we have a `@final` in it (but not declare really as a final class). So this worked well, but Phpstan doesn't pass.

To me, I prefer to extends the symfony class and hide this error in phpstan, but It could be tricky if in minor update they change the declaration of the class.

What do you prefer about it?

Reply:
Done anyway! 😄